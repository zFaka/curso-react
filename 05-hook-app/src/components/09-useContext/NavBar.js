import React from 'react';
import {NavLink, Link} from 'react-router-dom';

const NavBar = () => {
  return (
    <>
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
          <Link exact activeClassName="active"  to='/' className="navbar-brand" >useContext</Link>
        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink exact activeClassName="active" to='/' className="nav-link active" >Home</NavLink>
            <NavLink exact activeClassName="active" to='/about' className="nav-link" >About</NavLink>
            <NavLink exact activeClassName="active" to='/login' className="nav-link" >Login</NavLink>
          </div>
        </div>
      </nav>
    </>
  )}

export default NavBar;
