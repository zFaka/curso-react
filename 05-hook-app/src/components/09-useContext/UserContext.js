import {createContext} from 'react';
createContext(null);
export const UserContext = createContext(null);
