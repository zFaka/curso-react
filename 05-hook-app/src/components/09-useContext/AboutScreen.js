import React, {useContext} from 'react';
import {UserContext} from './UserContext';

const AboutScreen = () => {
  const {user, setUser} = useContext(UserContext);


  const handleClick = () => {

    setUser({})
  };


  return (
    <>
      <h1>About</h1> 
      <pre>
        {JSON.stringify(user, null, 3)}
      </pre>

      <button
        className='btn btn-primary'
        onClick={handleClick}
      >
        LogOut
      </button>
    </>
  )}

export default AboutScreen;
