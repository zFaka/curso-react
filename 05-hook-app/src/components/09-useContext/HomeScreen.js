import React, {useContext} from 'react';
import {UserContext} from './UserContext';

const HomeScreen = () => {
  const {user} = useContext(UserContext);
  console.log(user);
  return (
    <>
      <h1>Home Screen</h1> 
      <pre>
        {JSON.stringify(user, null, 3)}
      </pre>
    </>
  )}

export default HomeScreen;
