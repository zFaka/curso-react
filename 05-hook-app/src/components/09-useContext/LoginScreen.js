import React, {useContext} from 'react';
import {UserContext} from './UserContext';

const LoginScreen = () => {

  const {setUser} = useContext(UserContext);

  const inputedData = {

      id:123, 
    nick:'chucko'
  };


  return (
    <>
      <h1>Login</h1> 
      <button 
        className='btn btn-primary'
        onClick={()=>setUser(inputedData)}
      >
        Login
      </button>
    </>
  )}

export default LoginScreen;
