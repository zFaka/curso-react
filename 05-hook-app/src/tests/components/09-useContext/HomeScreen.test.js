import {mount} from "enzyme";
import HomeScreen from "../../../components/09-useContext/HomeScreen";
import {UserContext} from "../../../components/09-useContext/UserContext";

describe('Test HomeScreen', () => {

  const user = {

    name:'Faka', 
    email:'f@ka.com'

  };

  //user === user:user
  const wrapper = mount(
    <UserContext.Provider 
      value={{
        user
      }}
    >
      <HomeScreen/>
    </UserContext.Provider>
  );




  test('Should render', () => {
    expect(wrapper)
      .toMatchSnapshot()
  })

})
