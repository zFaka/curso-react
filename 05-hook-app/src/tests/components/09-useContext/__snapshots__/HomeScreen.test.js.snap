// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomeScreen Should render 1`] = `
Array [
  <h1>
    Home Screen
  </h1>,
  <pre>
    {
   "name": "Faka",
   "email": "f@ka.com"
}
  </pre>,
]
`;
